docs/source/progress.rst

```diff
@@ -199,6 +209,43 @@ If the :class:`~rich.progress.Progress` class doesn't offer exactly what you nee
         def get_renderables(self):
             yield Panel(self.make_tasks_table(self.tasks))
 
+Reading from a file
+~~~~~~~~~~~~~~~~~~~
+
+You can obtain a progress-tracking reader using the :meth:`~rich.progress.Progress.open` method by giving it a path. You can specify the number of bytes to be read, but by default :meth:`~rich.progress.Progress.open` will query the size of the file with :func:`os.stat`. You are responsible for closing the file, and you should consider using a *context* to make sure it is closed ::
+
+    import json
+    from rich.progress import Progress
+
+    with Progress() as progress:
+        with progress.open("file.json", "rb") as file:
+            json.load(file)
+
+
+Note that in the above snippet we use the `"rb"` mode, because we needed the file to be opened in binary mode to pass it to :func:`json.load`. If the API consuming the file is expecting an object in *text mode* (for instance, :func:`csv.reader`), you can open the file with the `"r"` mode, which happens to be the default ::
+
+    from rich.progress import Progress
+
+    with Progress() as progress:
+        with progress.open("README.md") as file:
+            for line in file:
+                print(line)
+
+
+Reading from a file-like object
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can obtain a progress-tracking reader wrapping a file-like object using the :meth:`~rich.progress.Progress.wrap_file` method. The file-like object must be in *binary mode*, and a total must be provided, unless it was provided to a :class:`~rich.progress.Task` created beforehand. The returned reader may be used in a context, but will not take care of closing the wrapped file ::
+
+    import io
+    import json
+    from rich.progress import Progress
+
+    with Progress() as progress:
+        file = io.BytesIO("...")
+        json.load(progress.read(file, total=2048))
```

Comment:
I think this example needs reworked - I personally don't understand what it's demonstrating. Since the paragraph immediately above the example refers to `wrap_file`, I would have expected this example to show me how to use that functionality.

Reply:
Fixed it, I forgot to update this one.